radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/RadixAtomValidator.java

```diff
@@ -69,8 +69,12 @@ public void validateSignatures(Atom atom) throws AtomValidationException {
 	@Override
 	public void validate(Atom atom) throws AtomValidationException {
 		// TODO: check with universe genesis timestamp
-		if (atom.getTimestamp() == null || atom.getTimestamp() == 0L) {
-			throw new AtomValidationException("Null or Zero Timestamp");
+		if (!atom.hasTimestamp()) {
+			throw new AtomValidationException("Atom does not contain a timestamp");
+		}
+
+		if (atom.getTimestamp() < 0L) {
```

Comment:
Either the comparator should be `<=`, or the error message should change here.  I think the `<=` is more consistent with previous checks.